mobile/src/main/java/butter/droid/ui/trailer/TrailerPlayerActivity.java

```diff
@@ -0,0 +1,208 @@
+import butter.droid.base.manager.youtube.YouTubeManager;
+import butter.droid.base.providers.media.models.Media;
+import butter.droid.base.torrent.StreamInfo;
+import butter.droid.base.torrent.TorrentService;
+import butter.droid.fragments.VideoPlayerFragment;
+import butter.droid.ui.ButterBaseActivity;
+import java.lang.ref.WeakReference;
+import java.net.URLDecoder;
+import javax.inject.Inject;
+import timber.log.Timber;
+
+public class TrailerPlayerActivity extends ButterBaseActivity implements TrailerPlayerView, VideoPlayerFragment.Callback {
+
+    public final static String LOCATION = "stream_url";
+    public final static String DATA = "video_data";
+
+    private static final String TAG = TrailerPlayerActivity.class.getSimpleName();
+
+    @Inject
+    TrailerPlayerPresenter presenter;
+    @Inject
+    YouTubeManager youTubeManager;
+    @Inject
+    NetworkManager networkManager;
+    @Inject
+    PhoneManager phoneManager;
+
+    private VideoPlayerFragment videoPlayerFragment;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        MobileButterApplication.getAppContext()
+                .getComponent()
+                .trailerComponentBuilder()
+                .trailerModule(new TrailerPlayerModule(this))
+                .build()
+                .inject(this);
+
+        super.onCreate(savedInstanceState, R.layout.activity_videoplayer);
+
+        final Media media = getIntent().getParcelableExtra(DATA);
+        media.title += " " + getString(R.string.trailer);
+        final String youtubeUrl = getIntent().getStringExtra(LOCATION);
+
+        this.videoPlayerFragment = (VideoPlayerFragment) getSupportFragmentManager().findFragmentById(R.id.video_fragment);
+
+        presenter.onCreate(media, youtubeUrl);
+    }
+
+    public static Intent getIntent(Context context, Media media, String url) {
+        Intent i = new Intent(context, TrailerPlayerActivity.class);
+        i.putExtra(TrailerPlayerActivity.DATA, media);
+        i.putExtra(TrailerPlayerActivity.LOCATION, url);
+        return i;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                finish();
+                return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public Long getResumePosition() {
+        return 0L;
+    }
+
+    @Override
+    public StreamInfo getInfo() {
+        return presenter.getStreamInfo();
+    }
+
+    @Override
+    public TorrentService getService() {
+        return null;
+    }
+
+    @Override
+    public void onDisableVideoPlayerSubsButton() {
+        videoPlayerFragment.enableSubsButton(false);
+    }
+
+    @Override
+    public void onExecuteQueryYoutubeTask(String youtubeUrl) {
+        final QueryYouTubeTask youTubeTask = new QueryYouTubeTask(this, youTubeManager, networkManager, phoneManager);
```

Comment:
Presenter is not for strictly for separating android and non android stuff. It is mainly for decoupling everything from activities/fragments. Yes you won't be able to test it in `tests` but you will instead have to use `androidTests` on the other hand if you leave it in activity you will still need to test it in `androidTests` but you will have to startup activity and it will be really hard to test ti.